### PR TITLE
Migrate OpenVINO inference to modelAPI

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## Description

WIP

- The openvino inference script can now be used standalone.
- Needs https://github.com/openvinotoolkit/anomalib/pull/1219 to be merged first

## TODO
- [ ] Update change log
- [ ] update tests

## Output
image

## Changes

<details>
<summary>Describe the changes you made</summary>

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

</details>

## Checklist

<details>
<summary>Ensure that you followed the following</summary>

- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas (If applicable)
- [ ] I have made corresponding changes to the documentation (If applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (If applicable)

</details>

### CODE REVIEW

File: tools/inference/openvino_inference.py [hunk 1]
```diff
@@ -4,16 +4,145 @@
 file system, and show the visualization results.
 """
 
-# Copyright (C) 2022 Intel Corporation
+# Copyright (C) 2023 Intel Corporation
```
Comment (samet-akcay) [inline tools/inference/openvino_inference.py:7]: should we do something like this
```suggestion
# Copyright (C) 2022-2023 Intel Corporation
```

File: tools/inference/openvino_inference.py [hunk 2]
```diff
@@ -4,16 +4,145 @@
...
+
+        Args:
+            anomaly_map (np.ndarray): Anomaly map.
+            image (np.ndarray): Input image.
+
+        Returns:
+            np.ndarray: Superimposed image.
+        """
+        # convert to color map
+        anomaly_map *= 255
```
Comment (samet-akcay) [inline tools/inference/openvino_inference.py:61]: should we have check here for the pixel range?

File: tools/inference/openvino_inference.py [hunk 3]
```diff
@@ -4,16 +4,145 @@
...
+                (0, 0, 255),
+                2,
+                cv2.LINE_AA,
+            )
+        if outputs.pred_boxes is not None:
+            self.superimpose_pred_boxes(image, outputs.pred_boxes)
+        return image
+
+
+class ImageReader:
```
Comment (samet-akcay) [inline tools/inference/openvino_inference.py:121]: this could also be on the map side

**Review by samet-akcay — CHANGES_REQUESTED** [review]
thanks, for this p!  just minor stuff

File: tools/inference/openvino_inference.py (hunk 2 above)
Comment (ashwinvaidya17) [reply tools/inference/openvino_inference.py:61]: Good idea

File: tools/inference/openvino_inference.py (hunk 1 above)
Comment (ashwinvaidya17) [reply tools/inference/openvino_inference.py:7]: Actually, I am not sure about the convention here. I am not even sure if people generally increment the number as well.

**Comment by ashwinvaidya17** [comment]
Closing as we need to rethink modelAPI dependency.

**Pull request closed without merging**